include/ginkgo/core/matrix/dense.hpp

```diff
@@ -130,6 +140,9 @@ class Dense
     friend class SparsityCsr<ValueType, int32>;
     friend class SparsityCsr<ValueType, int64>;
     friend class Dense<to_complex<ValueType>>;
+    friend class distributed::Vector<ValueType, int32, int32>;
+    friend class distributed::Vector<ValueType, int32, int64>;
+    friend class distributed::Vector<ValueType, int64, int64>;
```

Comment:
We are storing the local vector by value in vector, so I needed that. But I can do without, although the initialization of the local vector now looks a bit odd.

Reply:
Right sorry, I remember writing that now :)

Reply:
This gives me a bit more trouble than expected. I tried to do `local_{*local_vector_type::create(...)` in the constructor, but that only partially works. Somehow, the executor of the intermediate vector is not copied. Is that a bug or expected?

Also, I don't think that I can switch to using `local_` as a pointer type, since both unique and shared will mess up cloning.

Reply:
as a friend class, you should be able to use the constructor directly, what do you need create(...) for?

Reply:
Yes, sorry I was thinking about ways how to do it without the friend declaration, but I think there is no way around it.